include/nv/target

```diff
@@ -18,6 +18,11 @@
 #  define _NV_COMPILER_NVCC
 #elif defined(__NVCOMPILER) && __cplusplus >= 201103L
 #  define _NV_COMPILER_NVCXX
+#elif defined(__clang__) && defined(__CUDA__) && defined(__CUDA_ARCH__)
+// clang compiling CUDA code, device mode.
+#  define _NV_COMPILER_CLANG_CUDA
+// Also define NVCC to enable the same code paths.
```

Comment:
If we did this the other possible way and changed all the places we enter the nvcc code path to also check for clang, how big would the diff be?